corehq/apps/app_manager/tests/test_real_build.py

```diff
@@ -0,0 +1,40 @@
+import json
+import os
+import tempfile
+
+import requests
+from django.test import SimpleTestCase
+from requests.auth import HTTPDigestAuth
+
+from corehq.apps.app_manager.models import Application
+
+
+class TestRealBuild(SimpleTestCase):
+    @property
+    def username(self):
+        return os.environ['TRAVIS_HQ_USERNAME']
+
+    @property
+    def password(self):
+        return os.environ['TRAVIS_HQ_PASSWORD']
+
+    def fetch_and_build_app(self, domain, app_id):
+        source_path = os.path.join(tempfile.gettempdir(), '{}.{}.json'.format(domain, app_id))
+        if os.path.isfile(source_path):
+            with open(source_path, 'r') as f:
+                app_source = json.load(f)
```

Comment:
Does this code branch ever run? It's a little bit dangerous to grab file content from the operating system shared temporary directory. A safer alternative is to use `tempfile.mkdtemp()` to securely create a private temporary directory and cache the path on the test instance. Also, looks like this is an optimization for future tests that have not been written yet, so possibly a case of YAGNI?